### PR TITLE
 Make the chairman test more robust 

### PR DESCRIPTION
**Pull request by dcoutts**
It is now structured in a much simpler way: we run the experiment and get a result. Then we inspect that result for two conditions that establish positive evidence of success. If either fails, we fail.

This way it is very clear that we are really testing for a positive condition. Previously it could fail to discover a consensus failure, and nevertheless exit with a successful exit code. This did not establish
positive evidence, and so could fail to detect certain classes of integration mistakes (like inability to connect to the nodes at all).

Also change the number of legacy / new node split from 4 : 3 to 1 : 6. This should help a bit with debugging problems with communication between the old and new nodes.

### CODE REVIEW

File: cardano-node/src/Cardano/Chairman.hs [hunk 1]
```diff
@@ -61,45 +65,254 @@ import           Ouroboros.Network.Protocol.Handshake.Version
...
+      , "while the mininum expected block number is ", show minBlockNo
+      ]
+
+-- | Progress is defined as each chain being at least of a certain length.
+--
+progressCondition :: BlockNo
+                  -> ConsensusSuccess blk
+                  -> Either (ProgressFailure blk) ()
+progressCondition minBlockNo (ConsensusSuccess _ tips) =
+    case filter (\(_,tip) -> Block.getTipBlockNo tip < At minBlockNo) tips of
```
Comment (coot) [inline cardano-node/src/Cardano/Chairman.hs]: why not use `find`?

File: cardano-node/src/Cardano/Chairman.hs [hunk 2]
```diff
@@ -61,45 +65,254 @@ import           Ouroboros.Network.Protocol.Handshake.Version
...
+                   -> Either (ConsensusFailure blk)
+                             (ConsensusSuccess blk)
+consensusCondition (SecurityParam securityParam) chains =
+    -- The (forkTooLong . chainForkPoints) predicate is not transitive.
+    -- As a consequence, we need to check it between all the pairs of chains:
+    let forks = [ ((peerid1, peerid2), chainForkPoints chain1 chain2)
+                | (peerid1, chain1) <- Map.toList chains
+                , (peerid2, chain2) <- Map.toList chains
+                ]
+     in case filter (forkTooLong . snd) forks of
```
Comment (coot) [inline cardano-node/src/Cardano/Chairman.hs]: Why not use `find`?

File: nix/nixos/cardano-cluster-service.nix [hunk 3]
```diff
@@ -51,25 +51,25 @@ in let
   ### Names and topologies
   legacy-topology       = svcLib.mkLegacyTopology
     { c-a-1 = { port = 3001;
-                static-routes = [ [ "c-a-2" ] [ "c-b-1" ] [ "c-b-2" ] [ "proxy" ] ];
+                static-routes = [ [ "proxy" ] [ "c-a-2" ] [ "c-b-1" ] ];
               };
       c-a-2 = { port = 3002;
-                static-routes = [ [ "c-b-1" ] [ "c-b-2" ] [ "c-a-1" ] ];
+                static-routes = [ [ "c-a-1" ] [ "c-b-2" ] [ "c-a-2" ] ];
```
Comment (dcoutts) [inline nix/nixos/cardano-cluster-service.nix]: This is connecting to itself.

File: nix/sources.json [hunk 4]
```diff
@@ -5,10 +5,10 @@
         "homepage": null,
         "owner": "input-output-hk",
         "repo": "cardano-byron-proxy",
-        "rev": "9bae138919c31b5aeac8f7244e2b28e5f31d9f4f",
-        "sha256": "0drl4hhmph729g5s2bds0j9sg71jzw3b2cai4cm439dn2xymwc9x",
+        "rev": "824efdae912531d92f29333b482d4d0fbf2f433a",
+        "sha256": "10s3ivzv6hb7mb32imi2dga02vlaibsx8z1m2h7ns8cfkj4yng30",
         "type": "tarball",
-        "url": "https://github.com/input-output-hk/cardano-byron-proxy/archive/9bae138919c31b5aeac8f7244e2b28e5f31d9f4f.tar.gz",
+        "url": "https://github.com/input-output-hk/cardano-byron-proxy/archive/824efdae912531d92f29333b482d4d0fbf2f433a.tar.gz",
```
Comment (dcoutts) [inline nix/sources.json]: We'll want to bump it to `5538821647728516d5702cbfd9fb3000fcf41082`, the 1.0.5 tag.

File: nix/sources.json [hunk 5]
```diff
@@ -17,10 +17,10 @@
         "homepage": "",
         "owner": "input-output-hk",
         "repo": "cardano-sl",
-        "rev": "d0cc2e6336e3c57771c775cafe831bc83db303d3",
-        "sha256": "1l0af54ivqs26ibsl4nih8800b99f99r1y33wcrnby28mjsph8s2",
+        "rev": "5b35df50cce1ba65da49848f9888c6324796f10f",
```
Comment (dcoutts) [inline nix/sources.json]: Yup. Looks correct.

**Pull request merged**